wasatch/FeatureIdentificationDevice.py

```diff
@@ -373,26 +412,161 @@ def wait_for_usb_available(self):
                 sleep(sleep_sec)
         self.last_usb_timestamp = datetime.datetime.now()
 
-    ##
-    # This function is provided to simulate random USB communication errors
-    # during regression testing, and is normally a no-op.
-    def check_for_random_error(self):
+    def _check_for_random_error(self) -> bool:
+        """
+        This function is provided to simulate random USB communication errors
+        during regression testing, and is normally a no-op.
+        """
         if not self.inject_random_errors:
             return False
 
         if random.random() <= self.random_error_perc:
             log.critical("Randomly-injected error")
-            self.schedule_disconnect(Exception("Randomly-injected error"))
+            self._schedule_disconnect(Exception("Randomly-injected error"))
             return True
         return False
 
     ##
-    # @returns True if believed successful
-    def send_code(self, bRequest, wValue=0, wIndex=0, data_or_wLength=None, label="", dry_run=False, retry_on_error=False, success_result=0x00):
-        if self.shutdown_requested or (not self.connected and not self.connecting):
-            log.debug("send_code: not attempting because not connected")
+    # Until support for even/odd InGaAs gain and offset have been added to the
+    # firmware, apply the correction in software.
```

Comment:
You can remove this comment.  We're permanently leaving even-odd InGaAs correction in software; there is no current plan to re-implement it in firmware.